lesson21/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+
+    for (const [userKey, userValue] of Object.entries(usersTableColumn)) {
+        const th = createAttributedElements({
+            tag:"th",
+            valuesByAttributes:{
+                class:"text-sm text-white px-6 py-4",
+            },
+            str:userValue
+        });
+
+        tr.appendChild(th);
+
+        addSortButtons(userKey,userValue,th);
+    };
+
+    table.appendChild(thead).appendChild(tr);
+}
+
+const renderTableBody = usersData => {
+    const tbody = document.createElement("tbody");
+    tbody.id = "js-usersTableBody";
+
+    for(const user of usersData){
+        const fragment = document.createDocumentFragment();
+        const tr = document.createElement("tr");
+
+        Object.keys(usersTableColumn).forEach((column) => {
+            const td = createAttributedElements({
+                tag:"td",
+                valuesByAttributes:{
+                    class:"border border-gray-300 px-4 py-2"
+                },
+                str:user[column]
+            })
+            fragment.appendChild(tr).appendChild(td);
+        });
+
+        table.appendChild(tbody).appendChild(fragment);
+    }
+}
+
+const sortButtons = [
+    {dataOrder:"asc",src:"../img/asc.svg", alt:"asc-image"},
+    {dataOrder:"desc",src:"../img/desc.svg",alt:"desc-image"},
+    {dataOrder:"both",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortByUsersTableColumn = {
+    "id" : "ID"
+}
+
+const addSortButtons = (userKey,userValue,th) => {
+    for(const value of Object.values(sortByUsersTableColumn)){
+        userValue === value && createSortButtons(userKey,th) ;
+    }
+}
+
+const createSortButtons = (columnKey,th) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtons){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-${columnKey}SortButton`,
+                'data-order':button.dataOrder
+            }
+        });
+
+        button.dataOrder !== "both" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+    th.appendChild(sortButtonsBox).appendChild(fragment);
+}
+
+const toggleHiddenClassSortButton = (sortButtons,index) => {
+    sortButtons[index].classList.add("hidden");
+    (index === sortButtons.length -1) ? index = 0 : index++;
+    sortButtons[index].classList.remove("hidden");
+}
+
+const upDateTableBody = usersData =>  {
```

Comment:
```suggestion
const updateTableBody = usersData =>  {
```
nit
Since it is one character in "UPDATE"

Reply:
You are right.
I fixed.
https://github.com/nakagawamai/js-lesson/pull/23/commits/4597325e30a5cafd322c10da2fa6cb5d2c2224b5